cli/src/main/java/com/devonfw/tools/ide/process/ProcessResultImpl.java

```diff
@@ -80,11 +90,8 @@ public void log(IdeLogLevel level, IdeContext context) {
   @Override
   public void log(IdeLogLevel outLevel, IdeContext context, IdeLogLevel errorLevel) {
 
-    if (!this.out.isEmpty()) {
-      doLog(outLevel, this.out, context);
-    }
-    if (!this.err.isEmpty()) {
-      doLog(errorLevel, this.err, context);
+    if (!this.outputMessages.isEmpty()) {
+      doLog(outLevel, getOutputMessages().stream().map(OutputMessage::message).collect(Collectors.toList()), context);
     }
```

Comment:
Are two separate log levels still necessary with a single lists of captures?

Reply:
There are special cases at least Eclipse for where we want to make err visible to the user but have out on debug that is hidden by default but can be enabled.
I would therefore recommend to:
1. first check if `outLevel == errorLevel` and then do as you suggest/implemented
2. otherwise simply use the old approach but then also first check if the log-level is actually enabled. So if it is disabled, we do nothing (not even calling `getOut()` or `getErr()` what would compute the extra `List`).

Reply:
done